src/runtime/parser.c

```diff
@@ -1643,11 +1643,16 @@ TSTree *ts_parser_parse(TSParser *self, const TSTree *old_tree, TSInput input) {
 
 TSTree *ts_parser_parse_string(TSParser *self, const TSTree *old_tree,
                                const char *string, uint32_t length) {
+  return ts_parser_parse_string_encoding(self, old_tree, string, length, TSInputEncodingUTF8);
+}
+
+TSTree *ts_parser_parse_string_encoding(TSParser *self, const TSTree *old_tree,
+                                        const void *string, uint32_t length, TSInputEncoding encoding) {
```

Comment:
Related to the above comment, I think it might actually be clearer to take a `const char *` pointer here. For applications that want to pass a UTF16 string represented as a different type like `const uint16_t *`, the caller would have to explicitly cast it to a `(const char *)`, but I think that's ok.

Reply:
Thanks! I wasn't sure the best way to handle that. I went with the `const void *` since it made it easier to deal with from Swift, but I'll update it here and above to use `const char *` everywhere and do the cast in Swift.